src/confusion_matrices.jl

```diff
@@ -533,15 +533,13 @@ end
 
 
 # ## Final method to do the computation
-
-function _confmat(ŷ, y, indexer, levels, ordered)
+function _confmat(ŷ, y, indexer::F, levels, ordered) where F
```

Comment:
I find it strange that this annotation makes any difference. I would have thought `f(x)` and `f(x::F) where F` where syntactically equivalent, no?

Reply:
@ablaom It makes a huge difference in this case. For the specific case of passing functions(also Types and Varargs) as arguments to other functions, Julia doesn't automatically specialize on the type of the function so it sacrifices a bit of runtime speed (almost insignificant) for a reduced memory footprint and increased compile time performance see [here](https://docs.julialang.org/en/v1/devdocs/functions/#compiler-efficiency-issues). The type parameter is needed if we want to force julia to specialize on the type of the function (i.e. our function that accepts another function is performance critical)